### ISSUE TITLE
pyclean module py3clean references debpython

### ISSUE THREAD

**Issue by StevanWhite** [bittner/pyclean#80]
In the `pyclean` package, the `py3clean` script has a dependency on `debpython`.
This doesn't look right to me, for a general-purpose script.

Running it, as-is, installed from PyPI, fails:

`pip install pyclean`
`py3clean '.'`

Traceback ends with

```
  File "/data/data/com.termux/files/usr/lib/python3.11/site-packages/pyclean/py3clean.py", line 32, in <module>
    from debpython import files as dpf
ModuleNotFoundError: No module named 'debpython'
```

Actually, in modern Pythons, py3clean should do the same as `pyclean`.
It was introduced in some Linux distros to distinguish cleanup of Python 3 from Python 2 directories.

Or... maybe it shouldn't be in the package at all.

Why this is a problem: I have older scripts which had to explicitly call
py3clean for the newer installations, and `pyclean` for the older ones.
Those scripts are now failing, due to the missing dependency.

**Comment by bittner** [comment]
The [Debian pyclean scripts](https://github.com/bittner/pyclean?tab=readme-ov-file#debian) are included for nostalgic reasons, only. Originally, my take was to have them included and maintained (improved) in what has become the `pyclean` Python package on PyPI. I've never managed to get in touch with the creators and maintainers, and discuss their use and future, though.

Technically, only the `modern` implementation is relevant. The 3 other modules (aka scripts) will be removed for version 3 of PyClean, [as mentioned](https://github.com/bittner/pyclean?tab=readme-ov-file#development) at the bottom of the README.

If you really care about the debpython dependency being included, so that we have the Debian scripts working without any external dependency, I'd be willing to include them and issue a last release with it (before version 3).

Looks like #1 is a related issue and points to the appropriate source files.

**Comment by bittner** [comment]
I'm running Ubuntu 22.04.03 LTS, and I have the `python3-debian` package installed, which came shipped by default. I would guess this is the same for other people also running a Debian distro.

The package contains the `debpython` Python package, which in turn makes `py3clean` work, because [the implementation ensures](https://salsa.debian.org/cpython-team/python3-defaults/-/blob/master/py3clean#L31) the modules is on the import path.

<details><summary>dpkg -L python3-debian (click to expand)</summary>
<p>

```console
$ dpkg -L python3-debian
/.
/usr
/usr/lib
/usr/lib/python3
/usr/lib/python3/dist-packages
/usr/lib/python3/dist-packages/deb822.py
/usr/lib/python3/dist-packages/debian
/usr/lib/python3/dist-packages/debian/__init__.py
/usr/lib/python3/dist-packages/debian/_deb822_repro
/usr/lib/python3/dist-packages/debian/_deb822_repro/__init__.py
/usr/lib/python3/dist-packages/debian/_deb822_repro/_util.py
/usr/lib/python3/dist-packages/debian/_deb822_repro/formatter.py
/usr/lib/python3/dist-packages/debian/_deb822_repro/parsing.py
/usr/lib/python3/dist-packages/debian/_deb822_repro/tokens.py
/usr/lib/python3/dist-packages/debian/_deb822_repro/types.py
/usr/lib/python3/dist-packages/debian/_util.py
/usr/lib/python3/dist-packages/debian/_version.py
/usr/lib/python3/dist-packages/debian/arfile.py
/usr/lib/python3/dist-packages/debian/changelog.py
/usr/lib/python3/dist-packages/debian/copyright.py
/usr/lib/python3/dist-packages/debian/deb822.py
/usr/lib/python3/dist-packages/debian/debfile.py
/usr/lib/python3/dist-packages/debian/debian_support.py
/usr/lib/python3/dist-packages/debian/debtags.py
/usr/lib/python3/dist-packages/debian/deprecation.py
/usr/lib/python3/dist-packages/debian/py.typed
/usr/lib/python3/dist-packages/debian/watch.py
/usr/lib/python3/dist-packages/debian_bundle
/usr/lib/python3/dist-packages/debian_bundle/__init__.py
/usr/lib/python3/dist-packages/python_debian-0.1.43+ubuntu1.1.egg-info
/usr/lib/python3/dist-packages/python_debian-0.1.43+ubuntu1.1.egg-info/PKG-INFO
/usr/lib/python3/dist-packages/python_debian-0.1.43+ubuntu1.1.egg-info/dependency_links.txt
/usr/lib/python3/dist-packages/python_debian-0.1.43+ubuntu1.1.egg-info/requires.txt
/usr/lib/python3/dist-packages/python_debian-0.1.43+ubuntu1.1.egg-info/top_level.txt
/usr/share
/usr/share/doc
/usr/share/doc/python3-debian
/usr/share/doc/python3-debian/HISTORY.deb822
/usr/share/doc/python3-debian/NEWS.Debian.gz
/usr/share/doc/python3-debian/README.rst
/usr/share/doc/python3-debian/changelog.gz
/usr/share/doc/python3-debian/copyright
/usr/share/doc/python3-debian/examples
/usr/share/doc/python3-debian/examples/changelog
/usr/share/doc/python3-debian/examples/changelog/changelog_to_file
/usr/share/doc/python3-debian/examples/changelog/simple_changelog
/usr/share/doc/python3-debian/examples/copyright
/usr/share/doc/python3-debian/examples/copyright/check_parse_and_dump.py
/usr/share/doc/python3-debian/examples/deb822
/usr/share/doc/python3-debian/examples/deb822/depgraph
/usr/share/doc/python3-debian/examples/deb822/grep-maintainer
/usr/share/doc/python3-debian/examples/deb822/grep_native_packages.py
/usr/share/doc/python3-debian/examples/deb822/render-dctrl
/usr/share/doc/python3-debian/examples/debfile
/usr/share/doc/python3-debian/examples/debfile/ar
/usr/share/doc/python3-debian/examples/debfile/changelog_head
/usr/share/doc/python3-debian/examples/debfile/dpkg-info
/usr/share/doc/python3-debian/examples/debfile/extract_cron
/usr/share/doc/python3-debian/examples/debtags
/usr/share/doc/python3-debian/examples/debtags/pkgwalk
/usr/share/doc/python3-debian/examples/debtags/reverse
/usr/share/doc/python3-debian/examples/debtags/smartsearch
/usr/share/doc/python3-debian/examples/debtags/tagminer
/usr/share/doc/python3-debian/examples/debtags/tagsbyrelevance
/usr/share/lintian
/usr/share/lintian/overrides
/usr/share/lintian/overrides/python3-debian
```

</p>
</details> 

IOW, py3clean assumes that you run a "standard Debian distro", which provides `python3-debian` by default. If you're not on Debian it's expected "normal" that you want to run the "modern" implementation, and not `--legacy`.

Read: _It only works on Debian. Period._

If that doesn't meet your expectation, please explain to let me understand better. (Thank you!)

**Comment by StevanWhite** [comment]
Hi, Peter,

The problem is, older scripts, which were working on an older distro, fail with an unhelpful error message.

It would be best to remove the py3clean and py2clean scripts altogether.

If they should be in the package scripts should remove the files and directories they were meant to remove.
It wouldn't be a huge job to make them work right.

Another option would be to have them print out a message saying what needs to be corrected in the calling script.

For example, I know that the python3 version of pyclean is implemented as a module, so once the python version is picked with an environment variable,
     `$(PYTHON) -m pyclean`
would do the same as py3clean on the older (debian) system, and also work on modern system that only implements Python 3.

But to leave them with a message suggesting the user should try to find a missing package is not good.

**Comment by bittner** [comment]
The solutions you ask for are all feasible. The easiest would be dropping the two additional entry point scripts (and the Debian scripts they refer to).

But before I take a decision, can you tell me more about your use case? So I can better understand what drives you. (Note, I know nothing about how my users actually use the software.)

- How and for which tasks do you use `pyclean`, `py2clean` and `py3clean`? (development, operations, etc.)
- How do you install pyclean? (Globally using ..., in a virtualenv, using Tox, ...)
- Do you care about having the ability to clean byte code from installed system packages? (Deb, RPM, etc.)
- Do you have any option you particularly like? (`--dry-run`, `--verbose`, `--erase`, etc.)
- Anything else?

**Comment by bittner** [comment]
I just released [PyClean 3.0.0](https://pypi.org/project/pyclean/#history), which drops Python 2.7 support and the 3 included Debian scripts (`py2clean`, `py3clean`, `pypyclean`).
